openmc/material.py

```diff
@@ -403,6 +405,41 @@ def add_nuclide(self, nuclide: str, percent: float, percent_type: str = 'ao'):
 
         self._nuclides.append(NuclideTuple(nuclide, percent, percent_type))
 
+    def add_elements_or_nuclides(self, components: dict):
+        """ Add multiple elements or nuclides to a material
+
+        Parameters
+        ----------
+        components : dict of str to tuple
+            Dictionary mapping element or nuclide names to a tuple containing
+            their atom or weight percent, as well as any other arguments.
+
+        Examples
+        --------
+        >>> mat = openmc.Material()
+        >>> components  = {'Li': (1.0, {'enrichment': 60.0, 'enrichment_target': 'Li7'}),
+        >>>               'Fl': 1.0,
+        >>>               'Be6': (0.5, 'wo')}
```

Comment:
A few design thoughts. Although the Python API does allow you to mix multiple components with `ao` or `wo`, it won't actually work on the C++ side (has to be all `ao` or all `wo`). In light of that, maybe `percent_type` should just be handled by a separate argument to this method, i.e.
```Python
material.add_elements_or_nuclides({'H': 2.0, 'O': 1.0}, percent_type='ao')
```

Also, for the dictionary option, it might be cleaner to put _all_ arguments into the dictionary rather than keep the numerical value separate, i.e.
```Python
components = {'Li': {'percent': 1.0, 'enrichment': 60.0, 'enrichment_target': 'Li7'}}
```

Reply:
Agreed. It is done.